lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1055,18 +1055,40 @@ struct FIRStmtParser : public FIRScopedParser {
   /// a passive-typed value and return that.
   Value convertToPassive(Value input, Location loc);
 
+  /// Attach invalid values to every element of the value.
+  void emitInvalidate(Value val, Location loc, bool isDuplex) {
+    // Invalidate doesn't match the semantics of connect or partial connect, and
+    // so we have to manually do something like expand-connects. Invalid value
+    // needs to be connected to every field of a bundle which is flip, or
+    // every field in a duplex operation.
+    auto invalidType = val.getType().cast<FIRRTLType>();
+    if (invalidType.isa<AnalogType>()) {
+      // Analog types must be 'attach'ed, not connected.
+      auto invalidVal =
+          builder.create<InvalidValuePrimOp>(loc, invalidType.getPassiveType());
+      builder.create<AttachOp>(loc, ValueRange{val, invalidVal});
```

Comment:
I guess this is what the old code was doing, but I think analog can just be dropped. The SFC's `RemoveValidIf` pass is doing this currently.

Reply:
Wow I was wondering how this was supposed to work.  Does this mean any attach to an invalid value should be dropped? I haven't checked, but are we outputting incorrect attaches to 0 right now?

Reply:
Invalid values of Analog type get lowered to a wire with nothing connected to them.

Reply:
I started dropping the attaches.

Note to self: I need to fix the expand-whens initialization checking here, right now it would complain about an analog wire with no connections.